solutions/usaco-761.mdx

```diff
@@ -0,0 +1,102 @@
+---
+id: usaco-761
+source: USACO Bronze 2017 December
+title: Milk Measurement
+author: Ryan Chou
+---
+
+[Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_measurement_bronze_dec17.html)
+
+## Solution
+
+Since there are only $3$ cows and $N \leq 1000$, we can sort all of the entries by date and update the selected cow's milk value for each entry.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+<LanguageSection>
+<PySection>
+  
+```py
+import sys
+sys.stdin = open('measurement.in', 'r')
+sys.stdout = open('measurement.out', 'w')
+
+n = int(input())
+
+# Logs will store each entry in FJ's book.
+logs = []
+
+# Every cow starts off producing 7 gallons of milk each.
+bessie, elsie, mildred = 7, 7, 7
+
+"""
+Since all cows produce the same amount of milk,
+they'll all be on the display at the beginning.
+"""
+display = {'b', 'e', 'm'}
+
+"""
+This will store the past display,
+so we can judge whether FJ needs to change his display.
+"""
+tmp_display = {}
+ans = 0
+
+for _ in range(n):
+	time, name, addition = input().split()
+	logs.append([int(time), name, int(addition)])
+
+# Sort all logs by time.
+logs.sort()
+
+# For every entry,
+for i in logs:
+
+    # Get the cow who has changed their milk output.
+	if i[1] == 'Bessie':
+		bessie += i[2]
+	elif i[1] == 'Elsie':
+		elsie += i[2]
+	elif i[1] == "Mildred":
+		mildred += i[2]
+
+    # Find the maximum amount of milk that's being produced.
+	curmax = max(bessie, max(elsie, mildred))
+
+    # Duplicate the current display, this will be used to 
+    # see whether the top cows stay the same.
+	tmp_display = display.copy()
+
+    # If Bessie is producing the maximum,
+	if bessie == curmax:
+        # add her to the display.
+		display.add('b')
+	else:
+        # If she isn't producing the maximum and she's currently 
+        # in the display, remove her.
+		if 'b' in display:
+			display.remove('b')
+    
+    # Same for Elsie.
+	if elsie == curmax:
+		display.add('e')
+	else:
+		if 'e' in display:
+			display.remove('e')
+
+    # Same for Mildred.
+	if mildred == curmax:
+		display.add('m')
+	else:
+		if 'm' in display:
+			display.remove('m')
+    
+    # If the display has changed, FJ needs to update it.
```

Comment:
i like comments as much as the next coder
but this just seems a bit too bloated
here's a list of the unnecessary lines (including whitespace)
26, 27, 29, 45, 50, 52, 53, 55, 56 are the ones i see rn

Reply:
👍